src/app/shared/automation-editor/effect-editor/check-effect/check-effect.component.ts

```diff
@@ -0,0 +1,95 @@
+import {Component, OnInit} from '@angular/core';
+import {AbilityCheck} from '../../types';
+import {EffectComponent} from '../shared/EffectComponent';
+
+@Component({
+  selector: 'avr-check-effect',
+  templateUrl: './check-effect.component.html',
+  styleUrls: ['../shared.scss']
+})
+export class CheckEffectComponent extends EffectComponent<AbilityCheck> implements OnInit {
+  checkType: 'basic' | 'dc' | 'contest' = 'basic';
+
+  constructor() {
+    super();
+  }
+
+  ngOnInit() {
+    if (this.effect.contestAbility != null) {
+      this.checkType = 'contest';
+    } else if (this.effect.dc != null) {
+      this.checkType = 'dc';
+    }
+  }
+
+  onCheckTypeChange() {
+    if (this.checkType === 'basic') {
+      // basic checks: remove contestAbility, dc, contestTie
+      this.effect.contestAbility = undefined;
+      this.effect.dc = undefined;
+      this.effect.contestTie = undefined;
+    } else if (this.checkType === 'dc') {
+      // dc: remove contestAbility, contestTie
+      this.effect.contestAbility = undefined;
+      this.effect.contestTie = undefined;
+      // set dc attributes
+      this.effect.dc = this.effect.dc ?? '';
+      this.effect.success = this.effect.success ?? [];
+      this.effect.fail = this.effect.fail ?? [];
+    } else {
+      // contest: remove dc
+      this.effect.dc = undefined;
+      // set contest attributes
+      this.effect.contestAbility = this.effect.contestAbility ?? 'athletics'
+      this.effect.contestTie = this.effect.contestTie ?? 'success';
```

Comment:
I think this should be set to default to 'neither'

Check contests keep the status quo on a tie, RAW

Reply:
Hmm, I guess I wrote the spec from the mindset of contests being used in such a manner that the target winning maintains the status quo (e.g. starting a grapple, offensive class features, maintaining a grapple [since in a button scope you'd use target: parent to contest against the grappler, and if the grappler wins you remain grappled]).

I can change the default here, though the default would need updating in the docs and in the Automation Engine. Do you think it's worth changing?

Reply:
Ah, fair enough. I don't think it's a huge deal, not something worth going in and changing in multiple places.